server/src/test/java/io/crate/auth/UserAuthenticationMethodTest.java

```diff
@@ -21,25 +21,93 @@
 
 package io.crate.auth;
 
+import static io.crate.role.metadata.RolesHelper.getSecureHash;
+import static io.crate.role.metadata.RolesHelper.userOf;
+import static io.crate.testing.auth.RsaKeys.PRIVATE_KEY_256;
+import static io.crate.testing.auth.RsaKeys.PUBLIC_KEY_256;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
+import java.security.KeyFactory;
 import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.spec.EncodedKeySpec;
 import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+import java.time.Clock;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.Month;
+import java.time.ZoneOffset;
+import java.util.Base64;
 import java.util.Collection;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
 
 import org.elasticsearch.common.settings.SecureString;
 import org.elasticsearch.test.ESTestCase;
 import org.junit.Test;
 
+import com.auth0.jwk.Jwk;
+import com.auth0.jwk.JwkProvider;
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+
+import io.crate.role.JwtProperties;
 import io.crate.role.Role;
 import io.crate.role.Roles;
 import io.crate.role.SecureHash;
 import io.crate.role.metadata.RolesHelper;
 
 public class UserAuthenticationMethodTest extends ESTestCase {
 
+    public static final LocalDateTime EXPIRES_AT = LocalDate.of(2030, Month.JANUARY, 1).atStartOfDay();
+
+    /**
+     * Base64 encoded token, which represents header/payload shown below (signed by RsaKeys.PRIVATE_KEY_256).
+     * Header:
+     * {
+     *   "alg": "RS256",
+     *   "typ": "JWT",
+     *   "kid": "1"
+     * }
+     * Payload:
+     * {
+     *   "iss": "https://console.cratedb-dev.cloud/api/v2/meta/jwk/",
+     *   "username": "cloud_user",
+     *   "exp": 1893456000
+     * }
+     */
+    public static final String JWT_TOKEN = """
```

Comment:
How expensive is it to generate this on the fly?
Then we can control the expiration date and avoid the need for the `clockSupplier` in the JWTAuthenticationMethod. And we can also get rid of the `JWTVerifier.BaseVerification` casts in it.